src/main/java/controls/CalendarControl.java

```diff
@@ -0,0 +1,53 @@
+package controls;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+
+public class CalendarControl {
+
+    private final SelenideElement root;
+
+    private final DateTimeFormatter headerFormatter = DateTimeFormatter.ofPattern("MMMM yyyy");
+
+    public CalendarControl(SelenideElement root) {
+        this.root = root;
+    }
+
+    public void selectDate(LocalDate targetDate) {
+        YearMonth target = YearMonth.from(targetDate);
+
+        int attempts = 0;
+        int maxAttempts = 12;
+        while (attempts < maxAttempts) {
+            SelenideElement monthHeader = root.$$(".DayPicker-Month").first().$(".DayPicker-Caption");
+
+            String currentMonthYear = monthHeader.getText();
+            YearMonth current = YearMonth.parse(currentMonthYear, headerFormatter);
+
+            if (current.equals(target)) {
+                break;
+            } else if (current.isBefore(target)) {
+                SelenideElement nextButton = root.$("[aria-label='Next Month']");
```

Comment:
If you capture an element by its aria-label attribute, you locator could be broken if we switching language